sunpy/net/scraper.py

```diff
@@ -436,15 +374,19 @@ def _extract_files_meta(self, timerange, extractor, matcher=None):
         `list` of `dict`
             List of metadata info for all URLs.
         """
-        self.extractor = extractor
         urls = self.filelist(timerange)
         metalist = []
         for url in urls:
-            metadict = parse(extractor, url)
+            metadict = parse(self.pattern, url)
             if metadict is not None:
                 append = True
                 metadict = metadict.named
                 metadict['url'] = url
+                if 'month' not in metadict:
+                    if 'month_name' in metadict:
+                        metadict['month'] = datetime.strptime(metadict['month_name'], '%B').month
+                    elif 'month_name_abbr' in metadict:
+                        metadict['month'] = datetime.strptime(metadict['month_name_abbr'], '%b').month
```

Comment:
What client would hit this code block?

Reply:
I've only come across clients only having the `month_name_abbr` / %b representation and no `month` / %m representation [here](https://github.com/sunpy/sunpy/blob/813592baab6181288d8d50d6676d585374554960/sunpy/net/tests/test_scraper.py#L198).

Haven't come across any just for the `month_name` condition so far.

Reply:
Would it be possible to add a unit test for the scarper to ensure that this codeblock works as expected?